### PR TITLE
Update quarkus version checks to adapt to new SNAPSHOT versions

### PR DESCRIPTION
**Pull request by zakkak**
See
https://github.com/quarkusio/quarkus/blob/3c8c772dda9ed99674c9d8a6cd9cb4635d1ce5a2/build-parent/pom.xml#L8
and
https://github.com/quarkusio/quarkus/blob/bc0ca288d32ba2df212d8c4a2783c0bb21ca99a1/build-parent/pom.xml#L8

Relates to https://github.com/graalvm/mandrel/issues/501

### CODE REVIEW

File: testsuite/src/it/java/org/graalvm/tests/integration/PerfCheckTest.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ public void testQuarkusJSONParseOnce(TestInfo testInfo) throws IOException, Inte
             Files.createDirectories(Paths.get(appDir.getAbsolutePath(), "logs"));
             assertTrue(app.buildAndRunCmds.cmds.length > 1);
 
-            if (QUARKUS_VERSION.majorIs(3) || QUARKUS_VERSION.isSnapshot()) {
+            if (QUARKUS_VERSION.compareTo(new QuarkusVersion("3.0.0")) >= 0) {
```
Comment (jerboaa) [inline testsuite/src/it/java/org/graalvm/tests/integration/PerfCheckTest.java:142]: Consider extracting the `3.0.0` version into a constant.

**Review by jerboaa — APPROVED** [review]
Consider adding some tests for `QuarkusVersion` comparisons so that this keeps working. Looks fine to me, though.

File: testsuite/src/it/java/org/graalvm/tests/integration/PerfCheckTest.java (hunk 1 above)
Comment (zakkak) [reply testsuite/src/it/java/org/graalvm/tests/integration/PerfCheckTest.java]: Done

**Comment by zakkak** [comment]
> Consider adding some tests for QuarkusVersion comparisons so that this keeps working. Looks fine to me, though.

I extracted constant versions in constant fields in QuarkusVersion and also introduced tests for the class.

Thanks for the review @jerboaa

**Comment by jerboaa** [comment]
> I extracted constant versions in constant fields in QuarkusVersion and also introduced tests for the class.

Very nice, thank you!

**Pull request merged**